### ISSUE TITLE
DatePicker: Extra/unparsed characters found in date with 5 digit year

### ISSUE THREAD

**Issue by jepsar** [primefaces/primefaces#6646]
**Describe the defect**
While typing a date the component sometimes will replace the input value with key press values at the wrong location of the date. This made me end up with a 5 digit year (20201). When going back to that view to modify the date, it will give the "Extra/unparsed characters found in date" exception. I cannot use the component to change the date.

**Reproducer**
Unfortunately the showcase is giving other issues currently at https://www.primefaces.org/showcase/ui/input/datepicker/datePicker.xhtml

```
Uncaught TypeError: this.viewDate.getMonth is not a function
    at $.<computed>.<computed>.renderDateView (datepicker.js.xhtml?ln=primefaces&v=8.0.2-SNAPSHOT:1098)
    at $.<computed>.<computed>.renderDateView (jquery-plugins.js.xhtml?ln=primefaces&v=8.0.2-SNAPSHOT:144)
    at $.<computed>.<computed>.renderPanelElements (datepicker.js.xhtml?ln=primefaces&v=8.0.2-SNAPSHOT:1079)
    at $.<computed>.<computed>.renderPanelElements (jquery-plugins.js.xhtml?ln=primefaces&v=8.0.2-SNAPSHOT:144)
    at $.<computed>.<computed>.updateViewDate (datepicker.js.xhtml?ln=primefaces&v=8.0.2-SNAPSHOT:2157)
    at $.<computed>.<computed>.updateViewDate (jquery-plugins.js.xhtml?ln=primefaces&v=8.0.2-SNAPSHOT:144)
    at $.<computed>.<computed>.hideOverlay (datepicker.js.xhtml?ln=primefaces&v=8.0.2-SNAPSHOT:1805)
    at $.<computed>.<computed>.hideOverlay (jquery-plugins.js.xhtml?ln=primefaces&v=8.0.2-SNAPSHOT:144)
    at HTMLDocument.documentClickListener (datepicker.js.xhtml?ln=primefaces&v=8.0.2-SNAPSHOT:1814)
```

I was using 8.0,

```
  <p:datePicker id="#{id}"
                value="#{bean[id]}"
                yearNavigator="true"
                yearRange="#{app.yearRange(-80, 10)}"
                monthNavigator="true"
                required="#{required}"
                disabled="#{disabled}"
                placeholder="#{pattern}"
                pattern="#{pattern}"
                mindate="#{mindate}"
                showTime="#{showTime}"
                showIcon="true"
                showOnFocus="false">
``` 

with pattern "dd-MM-yyyy"

**Environment:**
 - PF Version: _8.0_
 - JSF + version: _Mojarra 2.3.9.payara-p3_
 - Affected browsers: _Chrome 87.0.4280.88 (Official Build) (x86_64)_

**To Reproduce**
Save a date with a 5 digit year

**Expected behavior**
Prevent 5 digit years.

**Comment by melloware** [comment]
I think this might already be fixed in 9.0.  When I try a 5 digit year in the latest showcase with 9.0.  I don't get any JavaScript error.

![image](https://user-images.githubusercontent.com/4399574/101917322-19dd5180-3b96-11eb-9712-ee88f804958a.png)

**Comment by melloware** [comment]
I am going to close this for now as its fixed.
